model/flow/ledger.go

```diff
@@ -70,4 +72,8 @@ func (d RegisterEntries) Values() []RegisterValue {
 type StorageProof = []byte
 
 // StateCommitment holds the root hash of the tree (Snapshot)
-type StateCommitment = []byte
+// TODO: solve the circular dependency and define StateCommitment as ledger.State
+type StateCommitment ledgerHash.Hash
```

Comment:
I think these are two different concepts, the idea was state commitment was a byte slice defined by the protocol, independent from any dependency to the ledger package. ledger package only has a notion of state, that specifically for flow protocol can be used as a source of state commitment, but it is possible that state commitments come from a different source or out come of f(commit) in the future, so I think we should define the fix size array here instead of relying on the same type and we do the type conversion whenever we want. to be honest I think that was the init reason to do things in slices, so conversions will be trivial.

Reply:
I understand what you mean. `StateCommitment` could be provided by different modules not only ledger. 
For now I can see that ledger is the only provider of state commitments, the type `StateCommitment` is defined itself in the `ledger` package. If we add a new provider of state commitments, I think it would be cleaner to:
 - define `StateCommitment` under `flow/model`, this way it is defined by the protocol not ledger.
 - define `State` in the ledger package and add a check that it can indeed be a `StateCommitment`.